Source/pack.cpp

```diff
@@ -289,45 +289,52 @@ void UnPackPlayer(const PlayerPack &packed, Player &player)
 
 	player = {};
 	player._pLevel = clamp<int8_t>(packed.pLevel, 1, MaxCharacterLevel);
+	player._pClass = static_cast<HeroClass>(clamp<uint8_t>(packed.pClass, 0, enum_size<HeroClass>::value - 1));
+
+	player._pBaseStr = std::min<uint8_t>(packed.pBaseStr, player.GetMaximumAttributeValue(CharacterAttribute::Strength));
+	player._pStrength = player._pBaseStr;
+	player._pBaseMag = std::min<uint8_t>(packed.pBaseMag, player.GetMaximumAttributeValue(CharacterAttribute::Magic));
+	player._pMagic = player._pBaseMag;
+	player._pBaseDex = std::min<uint8_t>(packed.pBaseDex, player.GetMaximumAttributeValue(CharacterAttribute::Dexterity));
+	player._pDexterity = player._pBaseDex;
+	player._pBaseVit = std::min<uint8_t>(packed.pBaseVit, player.GetMaximumAttributeValue(CharacterAttribute::Vitality));
+	player._pVitality = player._pBaseVit;
+	player._pStatPts = packed.pStatPts;
+
 	player._pMaxHPBase = SDL_SwapLE32(packed.pMaxHPBase);
+	player._pMaxHPBase = std::min<int32_t>(player._pMaxHPBase, player.calculateBaseLife());
 	player._pHPBase = SDL_SwapLE32(packed.pHPBase);
 	player._pHPBase = clamp<int32_t>(player._pHPBase, 0, player._pMaxHPBase);
 	player._pMaxHP = player._pMaxHPBase;
 	player._pHitPoints = player._pHPBase;
+
+	player._pMaxManaBase = SDL_SwapLE32(packed.pMaxManaBase);
+	player._pMaxManaBase = std::min<int32_t>(player._pMaxManaBase, player.calculateBaseMana());
+
+	player._pManaBase = SDL_SwapLE32(packed.pManaBase);
+	player._pManaBase = std::min<int32_t>(player._pManaBase, player._pMaxManaBase);
+	player._pMemSpells = SDL_SwapLE64(packed.pMemSpells);
+
 	player.position.tile = position;
 	player.position.future = position;
 	player.setLevel(clamp<int8_t>(packed.plrlevel, 0, NUMLEVELS));
 
-	player._pClass = static_cast<HeroClass>(clamp<uint8_t>(packed.pClass, 0, enum_size<HeroClass>::value - 1));
-
 	ClrPlrPath(player);
 	player.destAction = ACTION_NONE;
 
 	CopyUtf8(player._pName, packed.pName, sizeof(player._pName));
+	if (strlen(player._pName) == 0) {

```

Comment:
No need to calculate the length just to check if it's empty:

```suggestion
	if (player._pName[0] == '\0') {
```